app/src/androidTest/java/at/tugraz/ist/swe/DrawLineTest.java

```diff
@@ -0,0 +1,171 @@
+package at.tugraz.ist.swe;
+
+import android.graphics.Bitmap;
+import android.graphics.Color;
+import android.support.test.espresso.UiController;
+import android.support.test.espresso.ViewAction;
+import android.support.test.espresso.action.MotionEvents;
+import android.support.test.espresso.action.Swipe;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.util.Log;
+import android.view.MotionEvent;
+import android.view.View;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.zakariya.flyoutmenu.FlyoutMenuView;
+
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(AndroidJUnit4.class)
+public class DrawLineTest {
+
+    @Rule
+    public ActivityTestRule<MainActivity> activityTestRule = new ActivityTestRule<>(MainActivity.class);
+
+    @Test
+    public void testSelectedTool()
+    {
+        onView(withId(R.id.toolFlyoutMenu)).perform(click());
+        activityTestRule.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                int amount_items = activityTestRule.getActivity().toolFlyoutMenu.getAdapter().getCount();
+                for(int counter = 0; 0 < amount_items; counter++)
+                {
+                    FlyoutMenuView.MenuItem result = activityTestRule.getActivity().toolFlyoutMenu.getAdapter().getItem(counter);
+                    if(((FlyoutToolbar.MenuItemImage)result).getID() == R.drawable.ic_si_glyph_line_two_angle_point)
+                    {
+                        activityTestRule.getActivity().toolFlyoutMenu.setSelectedMenuItem(result);
+                        break;
+                    }
+                }
+                PaintingTool tool = activityTestRule.getActivity().drawingArea.getPaintingTool();
+                Log.d("TEST", tool.toString());

```

Comment:
Suggesting to remove the log. Not needed in a release version.

Reply:
Done.